### PR TITLE
Reset es password

### PR DESCRIPTION
**Pull request by tobio**
## Description

Adds a `reset_elasticsearch_password` attribute to the deployment resource, when true this explicitly resets the password, updating the value in state with the new credentials. 

## Related Issues

Fixes https://github.com/elastic/terraform-provider-ec/issues/99

## Motivation and Context

With the migration to 0.6 and higher requiring re-importing existing deployments, which in turn does not include the current elastic user, customers need a path to programatically update the credentials in TF state.  

## How Has This Been Tested?

Acceptance tests, manual tests. 

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/ecresource/deploymentresource/deployment/v2/schema.go [hunk 1]
```diff
@@ -132,3 +141,38 @@ func DeploymentSchema() tfsdk.Schema {
...
+	var isResetting *bool
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("reset_elasticsearch_password"), &isResetting)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if isResetting != nil && *isResetting {
+		planVal := resp.AttributePlan.(types.String)
+		planVal.Unknown = true
+		resp.AttributePlan = planVal
```
Comment (dimuon) [inline ec/ecresource/deploymentresource/deployment/v2/schema.go:176]:
```suggestion
		resp.AttributePlan = types.String{Unknown: true}
```

Comment (dimuon) [reply ec/ecresource/deploymentresource/deployment/v2/schema.go:176]: Sorry, the suggestion doesn't reflect the all proposed changes. `planVal` can be totally removed.

**Review by dimuon — APPROVED** [review]
👍

File: ec/ecresource/deploymentresource/deployment/v2/schema.go (hunk 1 above)
Comment (tobio) [reply ec/ecresource/deploymentresource/deployment/v2/schema.go:176]: All good. Good pickup, I was testing something out here and forgot to revert it.

**Pull request merged**